hack/generator/pkg/astmodel/conversion_function_builder.go

```diff
@@ -0,0 +1,494 @@
+		result.Body.List = append(result.Body.List, params.AssignmentHandler(params.ClonedDestination(), dst.Clone(actualDestination).(dst.Expr)))
+	}
+
+	return []dst.Stmt{result}
+}
+
+// IdentityAssignTypeName handles conversion for TypeName's that are the same
+// Note that because this handler is dealing with TypeName's and not Optional<TypeName>, it is safe to
+// perform a simple assignment rather than a copy.
+// This function generates code that looks like this:
+//// <destination> <assignmentHandler> <source>
+func IdentityAssignTypeName(_ *ConversionFunctionBuilder, params ConversionParameters) []dst.Stmt {
+	destinationType, ok := params.DestinationType.(TypeName)
+	if !ok {
+		return nil
+	}
+
+	sourceType, ok := params.SourceType.(TypeName)
+	if !ok {
+		return nil
+	}
+
+	// Can only apply basic assignment for typeNames that are the same
+	if !sourceType.Equals(destinationType) {
+		return nil
+	}
+
+	return []dst.Stmt{
+		params.AssignmentHandlerOrDefault()(params.ClonedDestination(), params.ClonedSource()),
+	}
+}
+
+// IdentityAssignPrimitiveType just assigns source to destination directly, no conversion needed.
+// This function generates code that looks like this:
+// <destination> <assignmentHandler> <source>
+func IdentityAssignPrimitiveType(_ *ConversionFunctionBuilder, params ConversionParameters) []dst.Stmt {
+	if _, ok := params.DestinationType.(*PrimitiveType); !ok {
+		return nil
+	}
+
+	if _, ok := params.SourceType.(*PrimitiveType); !ok {
+		return nil
+	}
+
+	return []dst.Stmt{
+		params.AssignmentHandlerOrDefault()(params.ClonedDestination(), params.ClonedSource()),
+	}
+}
+
+// IdentityAssignValidatedTypeDestination generates an assignment to the underlying validated type Element
+func IdentityAssignValidatedTypeDestination(builder *ConversionFunctionBuilder, params ConversionParameters) []dst.Stmt {
+	validatedType, ok := params.DestinationType.(*ValidatedType)
+	if !ok {
+		return nil
+	}
+
+	// pass through to underlying type
+	params = params.WithDestinationType(validatedType.ElementType())
+	return builder.BuildConversion(params)
+}
+
+// IdentityAssignValidatedTypeSource generates an assignment to the underlying validated type Element
+func IdentityAssignValidatedTypeSource(builder *ConversionFunctionBuilder, params ConversionParameters) []dst.Stmt {
+	validatedType, ok := params.SourceType.(*ValidatedType)
+	if !ok {
+		return nil
+	}
+
+	// pass through to underlying type
+	params = params.WithSourceType(validatedType.ElementType())
+	return builder.BuildConversion(params)
+}
+
+// IdentityDeepCopyJSON special cases copying JSON-type fields to call the DeepCopy method.
+// It generates code that looks like:
+//     <destination> = *<source>.DeepCopy()
+func IdentityDeepCopyJSON(builder *ConversionFunctionBuilder, params ConversionParameters) []dst.Stmt {
+
+	if !params.DestinationType.Equals(JSONType) {
+		return nil
+	}
+
+	newSource := &dst.UnaryExpr{
+		X: &dst.CallExpr{
```

Comment:
Use `astbuilder.CallQualifiedFunc()`

Reply:
`CallQualifiedFunc` expects that I have a `string` for `qualifier` but here I have an `Expr`. Given that seems to be an "abnormal" case (we have a good number of other usages of `CallQualifiedFunc`, and the fact that we have helpers for the other bits here I'm not sure it makes sense adding a helper to `astbuilder` to cater to this scenario.

The code I have right now is:
```
newSource := astbuilder.Dereference(
		&dst.CallExpr{
			Fun: astbuilder.Selector(params.ClonedSource(), "DeepCopy"),
			Args: []dst.Expr{},
		})
```

which isn't too bad?

Reply:
Yeah, that's better than it was. 👍